peachjam/adapters/indigo.py

```diff
@@ -551,33 +552,45 @@ def create_publication_file(self, publication_document, doc, title, stub=False):
                     logger.info("  Stub: No source file, skipping")
         else:
             url = publication_document["url"]
-            with NamedTemporaryFile() as f:
-                r = self.client_get(url)
-                filename = self.get_filename(r, f"Publication: {title}")
-                f.write(r.content)
-                mimetype = magic.from_file(f.name, mime=True)
-
-                if publication_document.get("has_trusted_url"):
-                    logger.info(f"  Using publication file from trusted URL {url}")
-                    PublicationFile.objects.update_or_create(
-                        document=doc,
-                        defaults={
-                            "filename": filename,
-                            "mimetype": mimetype,
-                            "size": f.size,
-                            "url": url,
-                            "use_source_file": False,
-                        },
+            filename = publication_document.get(
+                "filename", f"{slugify('Publication: ' + title)}.pdf"
```

Comment:
the `filename` entry will always be there, but may be None. So you want something like this instead:

```python
filename = publication_document.get("filename") or slugify(...)
```

Reply:
thanks! how embarrassing, I thought the second argument would be returned if get() returned None 🙈